### PR TITLE
Fix 2D instance drop position

### LINKED ISSUE #26075: Drag & Dropping 2D scene onto non-2D nodes doesn't respect drop placement
**Godot version:**

aa5b99821b23d74eafb49f4b0d2d86fe693a903c

**OS/device including version:**

Arch Linux

**Issue description:**

When you drag and drop a scene with a CanvasItem root somewhere into the viewport, typically the node will have its position set appropriately where you drop it, assuming the currently selected node inherits CanvasItem.

However, when performing this action on any other type of node, such as a Node, an AnimationPlayer, or a CanvasLayer, then the CanvasItem scene will simply be placed at (0, 0), when it would be preferred that the scene be placed where the user dropped it.

Ideally, when a CanvasItem scene is dragged into the viewport and the node it will become a child to doesn't inherit CanvasItem, it will default instead to just the position of the cursor.

**Steps to reproduce:**
1. Create and save a scene which has a single sprite
2. Attempt to drag and drop this scene into another scene where the selected node in the Scene tab doesn't inherit CanvasItem
3. Observe that the instanced scene defaults to (0, 0), rather than where you dropped it.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #26075

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -5501,17 +5501,17 @@ bool CanvasItemEditorViewport::_create_instance(Node *parent, String &path, cons
...
-			target_pos += instance_ci->_edit_get_position();
-		}
-		editor_data->get_undo_redo().add_do_method(instantiated_scene, "set_position", target_pos);
 	}
+	// Preserve instance position of the original scene.
+	CanvasItem *instance_ci = Object::cast_to<CanvasItem>(instantiated_scene);
+	if (instance_ci) {
+		target_pos += instance_ci->_edit_get_position();
+	}
+	editor_data->get_undo_redo().add_do_method(instantiated_scene, "set_position", target_pos);
```
Comment (akien-mga) [inline editor/plugins/canvas_item_editor_plugin.cpp:5514]: What happens if the instantiated scene is not a CanvasItem and doesn't have `set_position`?

Comment (KoBeWi) [reply editor/plugins/canvas_item_editor_plugin.cpp:5514]: There is an error.
It was already here actually, if the parent was CanvasItem. I'll fix it.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**